README.md
## McKinley Theme ##
>v.1.3

Medium inspired theme for Statamic [by @jamescampbell](http://twitter.com/jamescampbell), see it in action over at James's site [jamescampbell.us](https://jamescampbell.us).

###Features:
* Responsive, mobile-first layout (not just mobile-friendly)
* two custom templates, page.html & post.html
* social links bar and copyright as a partial-type template.
* Full bleed background images that fill entire screen for each post (recommend no larger than 1000px width)
* Rotating random images background on homepage (set images 1.jpg,2.jpg,3.jpg,4.jpg in img folder)
* Title, Sub-Title, Date, word-count / read-time per blog post
* Categories, 404 page, and Archives support
* Set your Twitter, Instagram, and Tumblr info directly from the theme.yaml file and watch the footer update with the proper social icon links
* SEO friendly customize meta tags in the theme.yaml file
* FONT-AWESOME support built-in (version 4.3) and in use for the social icons bar

> Uses very minimal pure javascript to get height of window, otherwise vanilla HTML / CSS. NO BLOAT!

### Examples
> page.html
The header for every blog post is simple, set your title, subtitle(captioned), and background image(header_img) as well as categories as you wish:

```
---
title: Switching to Statamic from Squarespace
author: James Campbell
captioned: tl;dr Recurring charges and lack of customization.
header_img: switching.jpg
categories: 
	- writing
---
```
See what the theme looks like:
* [Blog Post Page](https://jamescampbell.us/blog/on-photography-and-joy)
* [Regular Page](https://jamescampbell.us/about)

### Installation
1. Clone this repo into your _themes directory or download and unzip to the _themes directory. Final structure should be _themes/mckinley.
2. Change you config in _config/settings.yaml to use the new theme by changing the active theme name to mckinley.
3. Change the theme.yaml file settings to match the way you want your site to look (metatags, profile photo link, images folder url, etc.)
4. Make especially sure to set your default blog post image folder properly as this will affect each post you make. Follow the example I have that is the way I structured mine which is under the /assets/img/blog folder with the title-post.jpg as the format. 
5. profit.

### Customization of settings

Here are some of the customizations copied directly from the theme.yaml file that are available:
```
# Let's get social!
# Set the following true or false to turn on and off bottom social icons
twitter_on: true
tumblr_on: true
instagram_on: true
feed_on: false
twitter_name: jamescampbell
tumblr_link: http://tumblr.jamescampbell.us
instagram_link: http://instagram.com/jamesanthonycampbell
feed_link: http://feed.j1c.us/jamescampbell

# copyright done right, always a good idea to have a copyright page, but you can set it to false to turn it off
copyright_on: true
copyright_info: All Rights Reserved James Campbell. All content subject to Copyright.
copyright_details_link: https://jamescampbell.us/copyright-info
```
##### Changelog available [here](changelog.md)

License
------------
The MIT License (MIT)


changelog.md
# Changelog

### Version 1.3
* Added in the Font-Awesome icons support for the social bar links

### Version 1.1
* Added variables everywhere that tie directly to the theme.yaml file so nothing is hard-coded in for you to mess with multiple files or have coding experience. 
* Set your twitter,instagram,tumblr,copyright info directly from the theme.yaml file
* Set your profile image on/off and the image path
* Set your meta keywords, description, site title, and site name on home page via theme.yaml as well

### Version 0.8
* Fixed bottom social media bar to be updated icons that turn black on hover.
* Fixed menu buttons to be black on hover.

### Version 0.5
* Removed 'about me' blurb partial from homepage and just left the avatar.
* General code cleanup.


js/mckinleymini.js
function clickIE(){return document.all?!1:void 0}function clickNS(e){return!(document.layers||document.getElementById&&!document.all)||2!=e.which&&3!=e.which?void 0:!1}function resize(){"undefined"!=typeof window.innerWidth?(viewportwidth=window.innerWidth,viewportheight=window.innerHeight):"undefined"!=typeof document.documentElement&&"undefined"!=typeof document.documentElement.clientWidth&&0!=document.documentElement.clientWidth?(viewportwidth=document.documentElement.clientWidth,viewportheight=document.documentElement.clientHeight):(viewportwidth=document.getElementsByTagName("body")[0].clientWidth,viewportheight=document.getElementsByTagName("body")[0].clientHeight),contained.style.height=viewportheight+"px",aboutsection.style.height=viewportheight+"px"}function overlay(){el=document.getElementById("overlay"),el.style.visibility="visible"==el.style.visibility?"hidden":"visible"}function randomimagehome(){var e=["1.jpg","2.jpg","3.jpg","4.jpg"],t=e[Math.floor(Math.random()*e.length)],n=document.getElementsByClassName("mastheader");n=n[0];var o="url(_themes/mckinley/img/"+t+")";n.style.backgroundImage=o}function scaleFont(){var e=window.innerWidth;e>1900?(document.body.className="",document.getElementsByTagName("body")[0].className+="extraWide"):e>1400?(document.body.className="",document.getElementsByTagName("body")[0].className+="wide"):e>1e3?(document.body.className="",document.getElementsByTagName("body")[0].className+="standard"):e>700?(document.body.className="",document.getElementsByTagName("body")[0].className+="narrow"):(document.body.className="",document.getElementsByTagName("body")[0].className+="extraNarrow"),setTimeout(scaleFont,100)}function smoothScroll(e){var t=e;do{if(t=t.parentNode,!t)return;t.scrollTop+=1}while(0==t.scrollTop);var n=0;do{if(e==t)break;n+=e.offsetTop}while(e=e.offsetParent);scroll=function(e,t,n,o){o++,o>30||(e.scrollTop=t+(n-t)/30*o,setTimeout(function(){scroll(e,t,n,o)},20))},scroll(t,t.scrollTop,n,0)}var message="Sorry, right-click has been disabled for your safety.";document.layers?(document.captureEvents(Event.MOUSEDOWN),document.onmousedown=clickNS):(document.onmouseup=clickNS,document.oncontextmenu=clickIE);var contained=document.getElementsByClassName("mastheader"),aboutsection=document.getElementsByClassName("about-me");aboutsection=aboutsection[0],contained=contained[0],document.oncontextmenu=new Function("return false"),document.onkeyup=function(e){27==e.keyCode?window.location="https://jamescampbell.us":187==e.keyCode&&overlay()},resize(),scaleFont(),randomimagehome();

js/mckinley.js


// turn off right click for everyone (be safe!)
var message="Sorry, right-click has been disabled for your safety.";

function clickIE() {if (document.all) {(message);return false;}}
function clickNS(e) {
	if (document.layers||(document.getElementById&&!document.all)) {
	if (e.which==2||e.which==3) {(message);return false;}}}
	if (document.layers){
			document.captureEvents(Event.MOUSEDOWN);document.onmousedown=clickNS;
		}
	else {
			document.onmouseup=clickNS;document.oncontextmenu=clickIE;
		}

// get client height and set it for background image and scroll panes
var contained = document.getElementsByClassName("mastheader");
var aboutsection = document.getElementsByClassName("about-me");
aboutsection = aboutsection[0] // only need the first one
contained = contained[0] // only need the first one

// set height for containers based on window height
function resize() {
    // the more standards compliant browsers (mozilla/netscape/opera/IE7) use window.innerWidth and window.innerHeight

if (typeof window.innerWidth != 'undefined') {
    viewportwidth = window.innerWidth,
    viewportheight = window.innerHeight
}

// IE6 in standards compliant mode (i.e. with a valid doctype as the first line in the document)

else if (typeof document.documentElement != 'undefined' && typeof document.documentElement.clientWidth != 'undefined' && document.documentElement.clientWidth != 0) {
    viewportwidth = document.documentElement.clientWidth,
    viewportheight = document.documentElement.clientHeight
}

// older versions of IE

else {
    viewportwidth = document.getElementsByTagName('body')[0].clientWidth,
    viewportheight = document.getElementsByTagName('body')[0].clientHeight
}
contained.style.height = viewportheight+"px";
aboutsection.style.height = viewportheight+"px";
}

document.oncontextmenu=new Function("return false");
document.onkeyup = (function(e) {
  if (e.keyCode == 27) { window.location = 'https://jamescampbell.us'; } // if esc key press
  else if (e.keyCode == 187) { overlay(); }// if equal sign pressed, display special message (see overlay function)
});

 function overlay() { // overlay easteregg when keycode pressed above
    el = document.getElementById("overlay");
   el.style.visibility = (el.style.visibility == "visible") ? "hidden" : "visible";
 }

function randomimagehome() {
    var images = ['1.jpg', '2.jpg', '3.jpg', '4.jpg']; // any image files you want to use here!
    var rando = images[Math.floor(Math.random() * images.length)]; // make the randomness!
    var classy = document.getElementsByClassName('mastheader');
    classy = classy[0];
    var urllink = 'url(_themes/mckinley/img/'+rando+')';
    classy.style.backgroundImage=urllink;
   };

function scaleFont() {

  var viewPortWidth = window.innerWidth;

  if (viewPortWidth > 1900) {
	  document.body.className = '';
	  document.getElementsByTagName('body')[0].className+='extraWide';
  }
  else if (viewPortWidth > 1400) {
	  document.body.className = '';
	  document.getElementsByTagName('body')[0].className+='wide';
	  }
  else if (viewPortWidth > 1000) {
	  document.body.className = '';
	  document.getElementsByTagName('body')[0].className+='standard';
  }
  else if (viewPortWidth > 700) {
	  document.body.className = '';
	  document.getElementsByTagName('body')[0].className+='narrow';
  }
  else {
	  document.body.className = '';
	  document.getElementsByTagName('body')[0].className+='extraNarrow';
  };
  setTimeout(scaleFont, 100);
}

resize();
scaleFont();
randomimagehome();


// FUCKING SMOOTH SCROLL TO LOCAL # REFERENCE LINK WITHOUT JQUERY
function smoothScroll(target) {
    var scrollContainer = target;
    do { //find scroll container
        scrollContainer = scrollContainer.parentNode;
        if (!scrollContainer) return;
        scrollContainer.scrollTop += 1;
    } while (scrollContainer.scrollTop == 0);

    var targetY = 0;
    do { //find the top of target relatively to the container
        if (target == scrollContainer) break;
        targetY += target.offsetTop;
    } while (target = target.offsetParent);

    scroll = function(c, a, b, i) {
        i++; if (i > 30) return;
        c.scrollTop = a + (b - a) / 30 * i;
        setTimeout(function(){ scroll(c, a, b, i); }, 20);
    }
    // start scrolling
    scroll(scrollContainer, scrollContainer.scrollTop, targetY, 0);
}


